lib/reporter.js

```diff
@@ -218,11 +218,11 @@ var CoverageReporter = function (rootConfig, helper, logger, emitter) {
     coverageMap.merge(result.coverage)
   }
 
-  this.onRunComplete = function (browsers, results) {
+  this.onRunComplete = async function (browsers, results) {
```

Comment:
Having this function async causes `result.exitCode` to be set too late to be taken into account when karma exits the process.
So when the coverage check thresholds are not met, the process still exits with code 0.

Reply:
I've created https://github.com/karma-runner/karma-coverage/issues/418 to track this issue